keel-artifact/src/test/kotlin/com/netflix/spinnaker/keel/artifact/ArtifactDeployingListenerTests.kt

```diff
@@ -0,0 +1,66 @@
+package com.netflix.spinnaker.keel.artifact
+
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.id
+import com.netflix.spinnaker.keel.events.ArtifactVersionDeploying
+import com.netflix.spinnaker.keel.test.DummyResourceSpec
+import com.netflix.spinnaker.keel.test.combinedMockRepository
+import com.netflix.spinnaker.keel.test.deliveryConfig
+import com.netflix.spinnaker.keel.test.resource
+import dev.minutest.junit.JUnit5Minutests
+import dev.minutest.rootContext
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+
+internal class ArtifactDeployingListenerTests : JUnit5Minutests {
+  class Fixture {
+    val realResource: Resource<DummyResourceSpec> = resource()
+    val r: Resource<DummyResourceSpec> = mockk()
```

Comment:
I think you can use a Spy to simplify this: https://github.com/mockk/mockk#spy

Reply:
oh thank goodness, i was trying to figure out how to do that but i didn't know the right words to search for!!! thank you!